config/peerpods/podvm/gcp-podvm-image-handler.sh

```diff
@@ -188,6 +202,44 @@ function create_image_from_prebuilt_artifact() {
   echo "GCP image created successfully from prebuilt artifact"
 }
 
+function image_exists() {
+    echo "Checking if GCP image exists"
+
+    local image_path latest_image_id
+
+    # This command returns something like this:
+    # https://www.googleapis.com/compute/v1/projects/my-project/global/images/my-image
+    # Use `cut` to the image name
+    image_path=$(gcloud compute images describe "${IMAGE_NAME}-${IMAGE_VERSION}" \
+        --project="${GCP_PROJECT_ID}" \
+        --format='get(selfLink)' 2>/dev/null | cut -d/ -f6-)
+
+    if [[ $? -ne 0 ]]; then
```

Comment:
Azure returns an error if image is missing, whereas AWS returns an empty response if image is missing.
What does GCP returns?

Reply:
Unfortunately, GCP doesn't return a specific exit code if the image is not present. It will show an error, but the return it will be 0. For that case we are going to hit one of the conditions following...